examples/discovery/discovery.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018-2020 opcua authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/debug"
+	"log"
+)
+
+func main() {
+	endpoint := flag.String("endpoint", "opc.tcp://localhost:4840", "OPC UA Endpoint URL")
+	flag.BoolVar(&debug.Enable, "debug", false, "enable debug logging")
+	flag.Parse()
+	log.SetFlags(0)
+
+	ctx := context.Background()
+
+	{
+		log.Println("Finding servers on network")
+		servers, err := opcua.FindServersOnNetwork(ctx, *endpoint)
+		if err != nil {
+			log.Printf("Error calling find servers on network: %v", err)
```

Comment:
Isn't this a `log.Fatal()`? Using `else` in Go is usually a sign that the code flow is wrong.

Reply:
Reason I didn't it that way was because from my understanding `FindServers` should be implemented by all servers and `FindServersOnNetwork` only by some. As I do both calls in the example I don't want to Fatal here as I don't see the second call then. One way would also be to reorder the calls.

Reply:
For reference

FindServers: https://reference.opcfoundation.org/Core/Part4/v104/docs/5.4.2
FindServersOnNetwork: https://reference.opcfoundation.org/Core/Part4/v104/docs/5.4.3
Quote: "Unlike [FindServers](https://reference.opcfoundation.org/search/17?t=FindServers), this [Service](https://reference.opcfoundation.org/search/17?t=Service) is only implemented by [Discovery](https://reference.opcfoundation.org/search/17?t=Discovery) [Servers](https://reference.opcfoundation.org/search/17?t=Servers)."